src/ghost.js

```diff
@@ -0,0 +1,186 @@
+const EventEmitter = require('events').EventEmitter
+const { decodeJWT, verifyJWT } = require('did-jwt')
+const Room = require('ipfs-pubsub-room')
+
+class GhostChat extends EventEmitter {
+
+  // TODO: map did to peer id
+  // TODO: for online list, what do we do about late joiners who don't know who's online yet
```

Comment:
I'm pretty sure we could use `ipfs.pubsub.ls()` to get a list of peerIDs. Then we could send request to each for getting their 3IDs. We can also use the same command to see if any user has left without calling the leaveChat function.

Reply:
This is what I did to solve it

```js
this._room.on('peer joined', (peer) => this.announce(peer))
```

Once a peer joins the room, our peer immediately announces its presence to the peer.